units/src/fee_rate.rs

```diff
@@ -94,20 +106,39 @@ impl FeeRate {
     /// rounded down.
     ///
     /// [`None`] is returned if an overflow occurred.
-    pub fn checked_mul_by_weight(self, rhs: Weight) -> Option<Amount> {
-        let sats = self.0.checked_mul(rhs.to_wu())?.checked_add(999)? / 1000;
+    pub const fn checked_mul_by_weight(self, rhs: Weight) -> Option<Amount> {
```

Comment:
Thanks for the review.  I've changed it so that the `return` statements are not needed, but I have left it as a `match` statement instead of `if let` so that the format of all of the `const` functions are similar.

Reply:
> I also wouldn't mind a TODO to grep for in the future 

I don't personally want TODO littered everywhere in the codebase (I know,  my bad, I added some myself recently). When we update MSRV we can just grep for 'pub const fn' and check them all.

Reply:
I'm just nit picking, although the nested match statements have ` None => None` where as the `if let` I posted above doesn't bother with an else branch which is cleaner imo.

> I don't personally want TODO littered everywhere in the codebase (I know, my bad, I added some myself recently). When we update MSRV we can just grep for 'pub const fn' and check them all.

sure